File: CHEFPTNT/answer.cpp
#include <stdio.h>
#include <string>
#include <iostream>
#include <math.h>

using namespace std;

int main() {

	int t;

	cin>>t;

	while(t --> 0) {
		int N, M, X, K;
		string s;

		cin>>N>>M>>X>>K>>s;

		int evenCount = 0, oddCount = 0;

		for(int i = 0 ; i < s.length() ; ++i) {
			if (s[i] == 'E') {
				++evenCount;
			}
			else {
				++ oddCount;
			}
		}
		
		if ((M == 0) || (X == 0) || (N > K)) {
			printf("no\n");
			continue;
		}

		int noOfEvenMonths = M/2;
		int noOfOddMonths = ceil(float(M)/2.0);

		if( N <= (min((noOfEvenMonths * X), evenCount) + min((noOfOddMonths * X), oddCount)) )
			printf("yes\n");
		else
			printf("no\n");
	}
}

File: CHEFPTNT/answer.py
import math

#Number of test cases
nTestCases = input()

#Loop through all the test cases
for x in range(0, nTestCases):

	noOfPatents, monthsOfYear, workersBandwidth, totalWorkers = map(int, raw_input().split())

	s = str(raw_input())

	evenCount = 0
	oddCount = 0

	for i in s:
		if i == 'E':
			evenCount += 1
		else:
			oddCount += 1

	if (monthsOfYear == 0) or (workersBandwidth == 0) or (noOfPatents > totalWorkers):
		print("no\n")
		continue

	noOfEvenMonths = monthsOfYear/2
	noOfOddMonths = math.ceil(float(monthsOfYear)/2.0)

#	print("noOfEvenMonths" + str(noOfEvenMonths), "noOfOddMonths" + str(noOfOddMonths), "workersBandwidth" + str(workersBandwidth), "evenCount" + str(evenCount), "oddCount" + str(oddCount))

	if noOfPatents <= (min((noOfEvenMonths * workersBandwidth), evenCount) + min((noOfOddMonths * workersBandwidth), oddCount)):
		print("yes\n")
	else:
		print("no\n")